clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -176,9 +176,15 @@ public CommitRequestManager(
      */
     @Override
     public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
-        // poll only when the coordinator node is known.
-        if (coordinatorRequestManager.coordinator().isEmpty())
+        // poll when the coordinator node is known and fatal errors are not present
+        if (coordinatorRequestManager.coordinator().isEmpty()) {
+            Optional<Throwable> fatalError = coordinatorRequestManager.fatalError();
+            if (fatalError.isPresent()) {
+                pendingRequests.unsentOffsetCommits.forEach(request -> request.future.completeExceptionally(fatalError.get()));
+                pendingRequests.unsentOffsetFetches.forEach(request -> request.future.completeExceptionally(fatalError.get()));
+            }
             return EMPTY;
```

Comment:
If there's an error, do we not want to also fail any `commitSync()`/`commitAsync()` calls? Or is that handled separately?

Reply:
> The CommitRequestManager does not clear the fatal error, but the AbstractHeartbeatRequestManager does. Can you explain and/or add some comments so that others can understand the distinction between these two cases?

I addressed some comments on this, If you have other question, Please let me know.